qiskit/primitives/backend_estimator_dev.py

```diff
@@ -0,0 +1,619 @@
+            backend.max_circuits = getattr(backend.configuration(), "max_experiments", None)
+        self._backend = backend
+
+    @property
+    def abelian_grouping(self) -> bool:
+        """Groups commuting observable components."""
+        try:
+            return self._abelian_grouping
+        except AttributeError:
+            return True
+
+    @abelian_grouping.setter
+    def abelian_grouping(self, abelian_grouping: bool) -> None:
+        if not isinstance(abelian_grouping, bool):
+            raise TypeError(
+                "Expected `bool` type for `abelian_grouping`, "
+                f"got `{type(abelian_grouping)}` instead."
+            )
+        self._abelian_grouping = abelian_grouping
+
+    @property
+    def skip_transpilation(self) -> bool:
+        """If `True`, transpilation of the input circuits is skipped."""
+        try:
+            return self._skip_transpilation
+        except AttributeError:
+            return False
+
+    @skip_transpilation.setter
+    def skip_transpilation(self, skip_transpilation: bool) -> None:
+        if not isinstance(skip_transpilation, bool):
+            raise TypeError(
+                "Expected `bool` type for `skip_transpilation`, "
+                f"got `{type(skip_transpilation)}` instead."
+            )
+        self._skip_transpilation = skip_transpilation
+
+    @property
+    def transpile_options(self) -> Options:
+        """Options for transpiling the input circuits."""
+        return self._transpile_options
+
+    def set_transpile_options(self, **fields):
+        """Set the transpiler options for transpiler.
+        Args:
+            **fields: The fields to update the options
+        """
+        self._transpile_options.update_options(**fields)
+
+    ################################################################################
+    ## IMPLEMENTATION
+    ################################################################################
+    # TODO: caching
+    def _run(
+        self,
+        circuits: tuple[QuantumCircuit, ...],
+        observables: tuple[BaseOperator | PauliSumOp, ...],
+        parameter_values: tuple[tuple[float, ...], ...],
+        **run_options,
+    ) -> PrimitiveJob:
+        job = PrimitiveJob(self._compute, circuits, observables, parameter_values, **run_options)
+        job.submit()
+        return job
+
+    def _compute(
+        self,
+        circuits: tuple[QuantumCircuit, ...],
+        observables: tuple[BaseOperator | PauliSumOp, ...],
+        parameter_values: tuple[tuple[float, ...], ...],
+        **run_options,
+    ) -> EstimatorResult:
+        """Solve expectation value problem on backend."""
+        # Pre-processing
+        circuit_bundles = self._preprocess(circuits, observables, parameter_values)
+        metadata_bundles = tuple(tuple(c.metadata for c in bun) for bun in circuit_bundles)
+        job_circuits = list(c for bun in circuit_bundles for c in bun)  # TODO: accept tuple
```

Comment:
this todo is simple; better to either do it (though I'm not sure why) or leave the comment.

Reply:
Oh, but it is not that simple because it is `Backend.run()` what does not accept `Sequences` other than `list`. Fixing this here is out of scope, hence the `TODO` (again, I'll open issues before merging and remove this from here).